hls-plugin-api/src/Ide/Types.hs

```diff
@@ -992,3 +1001,55 @@ getProcessID = fromIntegral <$> P.getProcessID
 
 installSigUsr1Handler h = void $ installHandler sigUSR1 (Catch h) Nothing
 #endif
+
+-- |When provided with both a codeAction provider and an affiliated codeAction
+-- resolve provider, this function creates a handler that automatically uses
+-- your resolve provider to fill out you original codeAction if the client doesn't
+-- have codeAction resolve support. This means you don't have to check whether
+-- the client supports resolve and act accordingly in your own providers.
+mkCodeActionHandlerWithResolve
```

Comment:
I wonder if we could also fold the handling of the `OwnedResolveData` into this, so that plugins don't have to see it. We could wrap the main handler to take out whatever's in `data`, wrap it in ORD, and send it; and then the second handler could strip it off agian.